deploy/docker/templates/nginx_app.conf.sh

```diff
@@ -7,67 +7,93 @@ CUSTOM_DOMAIN="$2"
 
 cat <<EOF
 server {
-  listen 80;
+  listen ${PORT:-80} default_server;
 $NGINX_SSL_CMNT  server_name $CUSTOM_DOMAIN ;
   client_max_body_size 100m;
 
-	gzip on;
-
-	root /opt/appsmith/editor;
-	index index.html index.htm;
-
-	location /.well-known/acme-challenge/ {
-		root /appsmith-stacks/data/certificate/certbot;
-	}
-
-	proxy_set_header X-Forwarded-Proto \$scheme;
-	proxy_set_header X-Forwarded-Host \$host;
-
-	location / {
-		try_files \$uri /index.html =404;
-
-		sub_filter __APPSMITH_SENTRY_DSN__ '\${APPSMITH_SENTRY_DSN}';
-		sub_filter __APPSMITH_SMART_LOOK_ID__ '\${APPSMITH_SMART_LOOK_ID}';
-		sub_filter __APPSMITH_OAUTH2_GOOGLE_CLIENT_ID__ '\${APPSMITH_OAUTH2_GOOGLE_CLIENT_ID}';
-		sub_filter __APPSMITH_OAUTH2_GITHUB_CLIENT_ID__ '\${APPSMITH_OAUTH2_GITHUB_CLIENT_ID}';
-		sub_filter __APPSMITH_MARKETPLACE_ENABLED__ '\${APPSMITH_MARKETPLACE_ENABLED}';
-		sub_filter __APPSMITH_SEGMENT_KEY__ '\${APPSMITH_SEGMENT_KEY}';
-		sub_filter __APPSMITH_OPTIMIZELY_KEY__ '\${APPSMITH_OPTIMIZELY_KEY}';
-		sub_filter __APPSMITH_ALGOLIA_API_ID__ '\${APPSMITH_ALGOLIA_API_ID}';
-		sub_filter __APPSMITH_ALGOLIA_SEARCH_INDEX_NAME__ '\${APPSMITH_ALGOLIA_SEARCH_INDEX_NAME}';
-		sub_filter __APPSMITH_ALGOLIA_API_KEY__ '\${APPSMITH_ALGOLIA_API_KEY}';
-		sub_filter __APPSMITH_CLIENT_LOG_LEVEL__ '\${APPSMITH_CLIENT_LOG_LEVEL}';
-		sub_filter __APPSMITH_GOOGLE_MAPS_API_KEY__ '\${APPSMITH_GOOGLE_MAPS_API_KEY}';
-		sub_filter __APPSMITH_TNC_PP__ '\${APPSMITH_TNC_PP}';
-		sub_filter __APPSMITH_VERSION_ID__ '\${APPSMITH_VERSION_ID}';
-		sub_filter __APPSMITH_VERSION_RELEASE_DATE__ '\${APPSMITH_VERSION_RELEASE_DATE}';
-		sub_filter __APPSMITH_INTERCOM_APP_ID__ '\${APPSMITH_INTERCOM_APP_ID}';
-		sub_filter __APPSMITH_MAIL_ENABLED__ '\${APPSMITH_MAIL_ENABLED}';
-		sub_filter __APPSMITH_DISABLE_TELEMETRY__ '\${APPSMITH_DISABLE_TELEMETRY}';
-		sub_filter __APPSMITH_RECAPTCHA_SITE_KEY__ '\${APPSMITH_RECAPTCHA_SITE_KEY}';
-		sub_filter __APPSMITH_RECAPTCHA_SECRET_KEY__ '\${APPSMITH_RECAPTCHA_SECRET_KEY}';
-		sub_filter __APPSMITH_RECAPTCHA_ENABLED__ '\${APPSMITH_RECAPTCHA_ENABLED}';
-	}
-
-	location /api {
-		proxy_pass http://localhost:8080;
-	}
-
-	location /oauth2 {
-		proxy_pass http://localhost:8080;
-	}
-
-	location /login {
-		proxy_pass http://localhost:8080;
-	}
-
-	location /rts {
-		proxy_pass http://localhost:8091;
-		proxy_http_version 1.1;
-		proxy_set_header Host \$host;
-		proxy_set_header Connection 'upgrade';
-		proxy_set_header Upgrade \$http_upgrade;
-	}
+  gzip on;
+
+  root /opt/appsmith/editor;
+  index index.html index.htm;
+
+  location /.well-known/acme-challenge/ {
+    root /appsmith-stacks/data/certificate/certbot;
+  }
+
+  location = /supervisor {
+    return 301 /supervisor/;
+  }
+
+  location /supervisor/ {
+    proxy_http_version 1.1;
+    proxy_buffering     off;
+    proxy_max_temp_file_size 0;
+    proxy_redirect     off;
+    proxy_set_header   Host             \$http_host/supervisor/;
```

Comment:
The `Host` header shouldn't contain the path right? The `/supervisor/` part? Shouldn't this be just `\$http_host;` instead? Like:

```suggestion
    proxy_set_header   Host             \$http_host;
```

Reply:
Hi @sharat87 this is configuration for setting host in the control request (action like stopping process, restarting process,...) from Supervisor UI. When performing the action, a request will be sent to supervisor to perform. Then, it return a URL to reload the UI. There is a case that if user use another port rather than 80, the `http_host` variable leads to the missing of the port in the Host in the header of request and it route user to the the wrong URL.

Reply:
Yeah, I understand the port path, which is why we are using `$http_host` instead of `$host`, since `$host` doesn't have port in it.

My question is regarding the _path_. The `/supervisor/` part. Host header value should never have the path. See the Host header syntax at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host#syntax, it can only contain the hostname and the port.